src/pybind/mgr/dashboard/frontend/cypress/integration/pools/pools.e2e-spec.ts

```diff
@@ -38,7 +38,7 @@ describe('Pools page', () => {
 
     it('should edit a pools placement group', () => {
       pools.exist(poolName, true);
-      pools.edit_pool_pg(poolName, 32);
+      pools.edit_pool_pg(poolName, 32, false);
```

Comment:
That `false` makes this work, as it skips the post-condition check (that the number of PGs modified exactly matches the displayed value). However, this is also reducing our 'real' coverage here.

Let me explain myself: the test coverage is not only about counting lines of code you ran during your testing, but ensuring the assertions in the tests succeeded. If we remove this check, we are letting the test go on without knowing if the outcome was ok.

My proposal for this fix would be to understand what's the actual PG status string, when not `${new_pg} active+clean`... If you check the run reported in the tracker, there's an screenshot:

![image](https://user-images.githubusercontent.com/37327689/111328258-5ffef100-866e-11eb-8a9c-23eff9917cc4.png)

If you zoom in, you'll see that the PG Status is `31 active+clean, 1 active+clean+scrubbing`, so in this case, this output should be considered valid as well, as the scrubbing is a process that we are not controlling (if we find it adds stability, we could stop it with `ceph osd set noscrub` flag).

My proposal here would be to process the output `{} active+clean, {} active+clean+scrubbing, {} active+clean+whatever...` and count only for the total number of PGs in `active+clean`, ignoring any possible third status (`scrubbing` or `whatever`).

Reply:
Yes, makes sense. But just want to confirm can the case be let's say 25 active+clean or it's just the scrubbing process for which we need to deal? As if that's the case I'm supposing that it will be always 
`1 active+clean+scrubbing`, right? And then we can just look for pg_num for active+clean to be atleast 31 (if `new_pg` = 32) , why make e2e tests complex (as I'm worried about the timeout issues if we do some more stuff here).

Reply:
@avanthakkar the reason for adding a wait that actively checks that the number of PGs are N and they all are in a given state is due to the fact that under some states other than `active+clean` the subsequent tests could fail...

That said, we could try something [like assertions with functions](https://docs.cypress.io/api/commands/should.html#Function) and a map-reduce function with a callback (in this case a [`reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) is enough):
```ts
  this.getTableRow(name).text().should(
	  (pgStatus) => {
        expect(pgStatus.reduce(
		  (acc, curr) =>
			  acc += curr.includes('active+clean') ? parseInt(curr.trim().split(' ')[0]) : 0, 0)
		  )}
	)
}
```

I tried the reduce and it works:
```js
> '31 active+clean, 1 active+clean+scrubbing'.split(',').reduce( (acc, obj) => acc += obj.includes('active+clean') ? parseInt(obj.trim().split(' ')[0]):0, 0)
32
```

Reply:
Yes it works! Thanks @epuertat , applied the changes.

Reply:
[`expect(...).to.equal(32)`](https://docs.cypress.io/guides/references/assertions.html#BDD-Assertions)?

Reply:
Already tried that..didn't work. It says `expected NaN  to equal 32`

Reply:
That rather sounds like the expression above returned a `NaN`. So a fix should be required in that expression?

Reply:
Yes, fixed the expression , seems to be working locally.